src/full/Agda/Syntax/Parser/Lexer.x

```diff
@@ -54,13 +54,14 @@ $nonalpha    = $idchar # $alpha
 $white_notab = $white # \t
 $white_nonl  = $white_notab # \n
 
-@number       = $digit+ | "0x" $hexdigit+
 @prettynumber = $digit+ ([_] $digit+)*
               | "0x" $hexdigit+ ([_] $hexdigit+)*
               | "0b" $binarydigit+ ([_] $binarydigit+)*
 @integer      = [\-]? @prettynumber
-@exponent     = [eE] [\-\+]? @number
```

Comment:
This old grammar was clearly broken.  While some language support hexfloats (e.g. Java and C99, see https://www.keil.com/support/man/docs/armcc/armcc_chr1359124238525.htm), the exponent is then not separated by `E/e` (which is a hexdigit itself) but by `P/p`.  So it is good to remove it.